Invoking npm.cmd on Windows is failing

When I run `cargo build` on Windows I get the following error:

```
error: failed to run custom build command for `web v0.1.0 (C:\repositories\rust\myproject\web)`

Caused by:
  process didn't exit successfully: `C:\repositories\rust\myproject\target\debug\build\web-b0554dea23c73a0f\build-script-build` (exit code: 101)
--- stdout
NPM_CMD is npm.cmd
 running npm install npm.cmd in "./assets"

--- stderr
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'C:\repositories\rust\myproject\web\assets\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'C:\repositories\rust\myproject\web\assets\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'C:\repositories\rust\myproject\web\assets\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module 'C:\repositories\rust\myproject\web\assets\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." }', web\build.rs:4:5
```

However, the same project compiles just fine on Linux.

The following code reproduced the same error:

```rust
    if let Err(e) = Command::new("npm.cmd")
        .arg("install")
        .current_dir(r#"c:\repositories\rust\myproject\web\assets\"#)
        .status()
    {
        eprintln!("Cannot execute install: {:?}", e);
    }
```
I think it is due to the fact that `npm.cmd` needs to set some variables and probably update PATH prior to running in the command shell. 

The following code seems to work without any problems.

```rust
    if let Err(e) = Command::new("cmd")
        .arg("/c")
        .arg("npm.cmd")
        .arg("install")
        .current_dir(r#"c:\repositories\rust\mdreader\web\assets\"#)
        .status()
    {
        eprintln!("Cannot execute install: {:?}", e);
    }
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -5,7 +5,7 @@ on: [push]
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    runs-on: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
     - uses: actions/checkout@v1

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -5,7 +5,10 @@ on: [push]
 jobs:
   build:
 
-    runs-on: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
 
     steps:
     - uses: actions/checkout@v1

--- a/src/impl.rs
+++ b/src/impl.rs
@@ -601,7 +601,11 @@ impl NpmBuild {
 
     #[cfg(windows)]
     fn command(&self) -> Command {
-        Command::new("cmd").arg("/c").arg(&self.executable)
+        let mut cmd = Command::new("cmd");
+
+        cmd.arg("/c").arg(&self.executable);
+
+        cmd
     }
 }
 

--- a/src/impl.rs
+++ b/src/impl.rs
@@ -554,7 +554,8 @@ impl NpmBuild {
 
     /// Executes `npm install`.
     pub fn install(self) -> io::Result<Self> {
-        if let Err(e) = Command::new(&self.executable)
+        if let Err(e) = self
+            .command()
             .arg("install")
             .current_dir(&self.package_json_dir)
             .status()
@@ -568,7 +569,8 @@ impl NpmBuild {
 
     /// Executes `npm run CMD`.
     pub fn run(self, cmd: &str) -> io::Result<Self> {
-        if let Err(e) = Command::new(&self.executable)
+        if let Err(e) = self
+            .command()
             .arg("run")
             .arg(cmd)
             .current_dir(&self.package_json_dir)
@@ -590,6 +592,16 @@ impl NpmBuild {
     /// Converts to `ResourceDir`.
     pub fn to_resource_dir(self) -> ResourceDir {
         self.into()
+    }
+
+    #[cfg(not(windows))]
+    fn command(&self) -> Command {
+        Command::new(&self.executable)
+    }
+
+    #[cfg(windows)]
+    fn command(&self) -> Command {
+        Command::new("cmd").arg("/c").arg(&self.executable)
     }
 }
 

